numpy/testing/nosetester.py

```diff
@@ -176,7 +176,18 @@ class NoseTester(object):
                 'pyrex_ext',
                 'swig_ext']
 
-    def __init__(self, package=None, raise_warnings=None):
+    def __init__(self, package=None, raise_warnings="release"):
+        # Back-compat: 'None' used to mean either "release" or "develop"
+        # depending on whether this was a release or develop version of
+        # numpy. These semantics are fine for testing numpy, but not so
+        # helpful for downstream projects like scipy that use
+        # numpy.testing. (They want to set this based on whether *they* are a
+        # release or develop version, not whether numpy is.) So we continue to
+        # accept 'None' for back-compat, but it's now just an alias for the
+        # default "release".
```

Comment:
This makes sense, except that the conditional below starting `if raise_warnings is None and` will never evaluate to `True` anymore, so I guess it can be removed.